pkg/autoscaler/autoscaler.go

```diff
@@ -345,6 +365,27 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 					expectedScaledMachineSet = machineSet
 				}
 			}
+
+			By("checking for the presence of the scale from zero annotations")
+			Eventually(func() (bool, error) {
+				//checking for the keys of the newly added upstream annotations from the CAO
+				for _, machineSet := range machineSets {
+					ms, err := framework.GetMachineSet(context.TODO(), client, machineSet.GetName())
+					Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSet %s", machineSet.GetName())
+
+					if ms.Annotations == nil {
+						return false, nil
+					}
+
+					Expect(ms.Annotations).To(HaveKey(annotationsutil.CpuKey))
+					Expect(ms.Annotations).To(HaveKey(annotationsutil.MemoryKey))
+					Expect(ms.Annotations).To(HaveKey(annotationsutil.GpuCountKey))
+					Expect(ms.Annotations).To(HaveKey(annotationsutil.GpuTypeKey))
+				}
+
+				return true, nil
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "No scale from zero annotations found")
```

Comment:
we don't strictly need to run this check again on another test, i would be happy with it on only one test, but since it's here i think it's fine to keep.